app/src/main/java/com/mapbox/vision/examples/utils/classification/SignResourceMapper.kt

```diff
@@ -2,55 +2,63 @@ package com.mapbox.vision.examples.utils.classification
-    fun getSignResourceForCurrentSpeed(uiSign: UiSign, speed: Float): Int
+    fun getSignResource(uiSign: UiSign, country: Country): Int
+    fun getSpeedSignResource(uiSign: UiSign, speed: Float, country: Country): Int
 
     class Impl(val context: Context) : SignResourceMapper {
 
-        private val numbersTypesArray = arrayOf(
-                UiSign.SignType.SpeedLimit,
-                UiSign.SignType.SpeedLimitEnd,
-                UiSign.SignType.SpeedLimitMin,
-                UiSign.SignType.SpeedLimitNight,
-                UiSign.SignType.SpeedLimitTrucks,
-                UiSign.SignType.SpeedLimitComplementary,
-                UiSign.SignType.SpeedLimitExit,
-                UiSign.SignType.SpeedLimitRamp,
-                UiSign.SignType.Mass
+        private val signsWithNumber = arrayOf(
+            UiSign.SignType.SpeedLimit,
+            UiSign.SignType.SpeedLimitEnd,
+            UiSign.SignType.SpeedLimitMin,
+            UiSign.SignType.SpeedLimitNight,
+            UiSign.SignType.SpeedLimitTrucks,
+            UiSign.SignType.SpeedLimitComplementary,
+            UiSign.SignType.SpeedLimitExit,
+            UiSign.SignType.SpeedLimitRamp,
+            UiSign.SignType.Mass
         )
 
-        private val overSpeedArray = arrayOf(
-                UiSign.SignType.SpeedLimit,
-                UiSign.SignType.SpeedLimitNight,
-                UiSign.SignType.SpeedLimitComplementary,
-                UiSign.SignType.SpeedLimitTrucks
+        private val speedLimitSigns = arrayOf(
+            UiSign.SignType.SpeedLimit,
+            UiSign.SignType.SpeedLimitNight,
+            UiSign.SignType.SpeedLimitComplementary,
+            UiSign.SignType.SpeedLimitTrucks
         )
 
-        private fun getResourceNameForSign(uiSign: UiSign) = if (uiSign.signType in numbersTypesArray) {
-            uiSign.signType.usResourceName + uiSign.signNum.value
-        } else {
-            uiSign.signType.usResourceName
-        }
-
-        private fun getResourceId(name: String) = context.resources.getIdentifier(
-                name, "drawable", context.packageName
-        ).also {
-            if (it == 0) {
+        private fun getResourceNameForSign(
+            uiSign: UiSign,
+            country: Country
+        ): String {
+            val resourceName = when (country) {
+                Country.Unknown, Country.USA -> uiSign.signType.usResourceName
+                Country.UK -> uiSign.signType.ukResourceName
+                Country.China, Country.Other -> uiSign.signType.chinaResourceName
+            }
 
+            return if (uiSign.signType in signsWithNumber) {
+                resourceName + uiSign.signNum.value
+            } else {
+                resourceName
             }
         }
 
-        override fun getSignResource(uiSign: UiSign) = getResourceId(
-                name = getResourceNameForSign(uiSign)
+        private fun getResourceId(name: String) = context.resources.getIdentifier(
```

Comment:
I think it should be an extension smt like `Context.getResourceId(@ResId resId, enum: ResourceType = Drawable)`
where enum:
enum ResourceType(val name: String){
  Drawable("drawable"),
  String("string")
  ...
}